src/genome.py

```diff
@@ -170,11 +201,42 @@ class LinkedListGenome(Genome):
     Implements the Genome interface using linked lists.
     """
 
-    def __init__(self, n: int):
-        """Create a new genome with length n."""
-        ...  # FIXME
+    active_TE = dict
+    te_id = int
 
-    def insert_te(self, pos: int, length: int) -> int:
+    def __init__(self, n: int):
+        self.active_TE = {}
+        self.next_TE_id = 1
+
+        # init LL Head
+        self.head = Node(0)
+        # self.head.prev = self.head
+        # seriously dont use below, this kills the program.
+        # self.head.next = self.head
+
+        # make genome
+        for _ in range(n):
+            self.insert(0)
+        self.length = n
+        # burde kunne "cirkuleres" hvis tail bare peger tilbage på head.
+
+    # burde nok ikke gøres uden at lave stops så while loops.
+
+    def insert(self, te):
+        insert = Node(te)
+        # no need to head check as head will allways start as 0, i hope.
+        current = self.head
+        while current.next:
+            current = current.next
+        current.next = insert
```

Comment:
Do you really want to search through the entire genome to get to the end, just to append? Does it matter whether you append or prepend?